x-pack/plugins/stack_connectors/common/openai/schema.ts

```diff
@@ -28,20 +28,64 @@ export const RunActionParamsSchema = schema.object({
   body: schema.string(),
 });
 
-// Run action schema
-export const InvokeAIActionParamsSchema = schema.object({
-  messages: schema.arrayOf(
+const AIMessage = schema.object({
+  role: schema.string(),
+  content: schema.string(),
+  name: schema.maybe(schema.string()),
+  function_call: schema.maybe(
     schema.object({
-      role: schema.string(),
-      content: schema.string(),
+      arguments: schema.string(),
+      name: schema.string(),
     })
   ),
+  tool_calls: schema.maybe(
+    schema.arrayOf(
+      schema.object({
+        id: schema.string(),
+        function: schema.object({
+          arguments: schema.string(),
+          name: schema.string(),
+        }),
+        type: schema.string(),
+      })
+    )
+  ),
+  tool_call_id: schema.maybe(schema.string()),
+});
+
+// Run action schema
+export const InvokeAIActionParamsSchema = schema.object({
+  messages: schema.arrayOf(AIMessage),
   model: schema.maybe(schema.string()),
+  functions: schema.maybe(
+    schema.arrayOf(
+      schema.object(
+        {
+          name: schema.string(),
+          description: schema.string(),
+          parameters: schema.object({
```

Comment:
this can be a very compllicated object, probably just want `schema.object({ }, { unknowns: 'allow' }}`